tests/test_utils_strict_dataclass.py

```diff
@@ -29,6 +29,24 @@ def strictly_positive(value: int):
         raise ValueError(f"Value must be strictly positive, got {value}")
 
 
+def dtype_validation(value):
+    "Torch isn't installed in runners so we will check only for `string`"
+    if not isinstance(value, str):
+        raise ValueError(f"Value must be string ot `torch.dtype` but got {value}")
+
+    if isinstance(value, str) and value not in ["float32", "bfloat16", "float16"]:
+        raise ValueError(f"Value must be one of `[float32, bfloat16, float16] but got {value}")
+
+
+@strict
+@dataclass
+class ConfigForwardRef:
+    model_type: str
+    dtype: Union[ForwardRef("torch.dtype"), str] = validated_field(validator=[dtype_validation])
```

Comment:
```suggestion
    dtype: Union[ForwardRef("torch.dtype"), str] = validated_field(validator=[dtype_validation])
```

Since the dtype is a Union, we are not really testing the `ForwardRef`. I just tried the exact same test on `main` and it passes correctly which shouldn't be the case if the test was correct. That's why I suggested to test something like this:

```py
@strict
@dataclass
class ConfigWithForwardRef:
    explicitForwardRef: ForwardRef("torch.dtype")
    implicitForwardRef: "torch.dtype"
```

Not that your test case doesn't have to look like a "real use case". We just want to test that the atomic validation works correctly